src/pages/index.js

```diff
@@ -74,25 +111,120 @@ const addImageModal = new ModalWithForms(
   config
 );
 
+// CREATE A MODALWITHFORM FOR CHANGE PICTURE
+const avatarEditModal = new ModalWithForms(
+  "#modal-change-picture",
+  handleAvatarFormSubmit,
+  config
+);
+
+
+// CREATE A MODALWITHFORM FOR DELETE CONFIRMATION
+const confirmationModal = new ModalWithConfirmation(
+  "#delete-confirmation-modal",
+{
+  submitButtonSelector: ".modal__button",
+  formSelector: ".modal__form",
+}
+);
+
+
+
 // FUNCTION PROFILE EDIT SUBMIT
-function handleProfileFormSubmit(values) {
-  userInfo.setUserInfo(values);
+function handleProfileFormSubmit(inputValues) {
+  profileEditModal.renderingSaving(true);
+  api
+    .editUserInfo(inputValues)
+    .then((res) => {
+      userInfo.setUserInfo({
+        name: res.name,
+        description: res.about,
+      });
+      profileEditModal.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      profileEditModal.renderingSaving(false);
+    });
   profileEditModal.close();
```

Comment:
Needs correction: Сlose modal in `then` block, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again.